Source/WebCore/css/color/CSSUnresolvedColorLayers.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "ColorInterpolationMethod.h"
+#include "StyleColor.h"
+#include <variant>
+#include <wtf/Forward.h>
+
+namespace WebCore {
+
+namespace Style {
+enum class ForVisitedLink : bool;
+}
+
+class CSSUnresolvedColor;
+class Document;
+class RenderStyle;
+enum class BlendMode : uint8_t;
+
+struct CSSUnresolvedColorResolutionContext;
+
+struct CSSUnresolvedColorLayers {
+    bool operator==(const CSSUnresolvedColorLayers&) const = default;
+
+    BlendMode blendMode;
+    Vector<UniqueRef<CSSUnresolvedColor>> colors;
```

Comment:
I haven't quite worked out how happy `Vector` would be about this, but could this be a `Vector<CSSUnresolvedColor>`? (it may be that Vector requires a fully defined type, in which case this would not work).

Reply:
It needs a fully defined type unfortunately.